sql/analyzer/check_auth.go

```diff
@@ -0,0 +1,31 @@
+package analyzer
+
+import (
+	"gopkg.in/src-d/go-mysql-server.v0/auth"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/plan"
+)
+
+// CheckAuthorizationRule is a rule that validates that a user can execute
+// the query.
+const CheckAuthorizationRule = "check_authorization"
+
+// CheckAuthorization creates an authorization check with the given Auth.
+func CheckAuthorization(au auth.Auth) RuleFunc {
```

Comment:
Could you delete the old rule that we used to make possible read-only queries?

Reply:
I'll add the possibility to specify the permissions when using `NewNativeSingle` so it can be used to disable writing instead of read only rule.

Reply:
Done